jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java

```diff
@@ -411,7 +416,13 @@ protected CommandOutput executeDeployGroupAppTemplate(final String groupName, fi
                 .setGroupName(groupName)
                 .setWebAppName(application.getName())
                 .build();
-        return resourceService.generateAndDeployFile(resourceIdentifier, application.getName(), fileName, hostName);
+        String lockKey = groupName + application.getName() + fileName + JwalaUtils.getHostAddress(hostName);
```

Comment:
We need to do this for the group deploy of the JVM and Web Server resources are well. Unfortunately these methods are also contained in the REST layer so you would want to refactor those methods into the service layer as well.